IsPalindrom/Program.cs

```diff
@@ -1,7 +1,4 @@
-﻿//Step 1 - create a new empty string "container"
-//step 2 - read the input string "backwards" and load the characters one by one into the new empty string "container"
-//step 3 - compare if the new string equals to the input string
-
+﻿// approach 1 with for loop
 bool isPalindrom (string stringToTest) 
```

Comment:
for the first isPalindrom method:

also good!

the only thing that comes to mind is that you are repeatedly allocating a new array in your for loop:

```

bool isPalindrom (string stringToTest) 
{ 
    string stringReversed = "";

    for (int i = stringToTest.ToCharArray().Length-1; i >= 0; i--)  <------- .ToCharArray once
    { 
        stringReversed += stringToTest.ToCharArray()[i]; <------ .ToCharArray i times
    }

    return stringToTest == stringReversed;
  
}
```

thats a lot of allocations. (i+1 times)

if you gonna use the char array this many time, I'd recommend you create a variable for it, and use that everywhere

Reply:
very good advise! This forces me to think about how the computer is gonna run the method in real world and to improve performance.